lib/middleware/middleware.go

```diff
@@ -0,0 +1,71 @@
+package middleware
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/kosyfrances/rundeck-zabbix/lib/utils"
+)
+
+// BuildRundeckURLEndpoint builds a URL endpoint given a baseURL and an endpoint.
+// It returns the full URL endpoint and an error if any.
+func BuildRundeckURLEndpoint(baseURL, endpoint string) (string, error) {
+	base, err := url.Parse(baseURL)
+	if err != nil {
+		return "", fmt.Errorf("cannot parse Rundeck base URL")
+	}
+
+	queryEndpoint, err := url.Parse(endpoint)
+	if err != nil {
+		return "", fmt.Errorf("cannot parse Rundeck query Endpoint")
+	}
+
+	return base.ResolveReference(queryEndpoint).String(), nil
+}
+
+/*
+GetRundeckJobID gets the job ID from Rundeck API.
+
+Note:
+Rundeck API does not return Host/Node name alongside when you try to get a job.
```

Comment:
Do you have a reference for this with the expected response from the API ? (I'll Google it :)) and come back to this otherwise the assumption might not be the best route (if there are other options)

Reply:
Here is the kind of response I got when I had multiple jobs with same name.
```
[{"id":"91c755ac-dab2-475c-b903-92e5ff59e740","name":"[RD] etc passwd has been changed on {HOST.NAME}","group":null,"project":"zabideck","description":"[RD] /etc/passwd has been changed on {HOST.NAME}","href":"http://localhost:4440/api/27/job/91c755ac-dab2-475c-b903-92e5ff59e740","permalink":"http://localhost:4440/project/zabideck/job/show/91c755ac-dab2-475c-b903-92e5ff59e740","scheduled":false,"scheduleEnabled":true,"enabled":true},

{"id":"826e88dc-926b-415f-a724-a3a847aae3f2","name":"[RD] etc passwd has been changed on {HOST.NAME}","group":null,"project":"zabideck","description":"[RD] /etc/passwd has bee
n changed on {HOST.NAME}","href":"http://localhost:4440/api/27/job/826e88dc-926b-415f-a724-a3a847aae3f2","permalink":"http://localhost:4440/project/zabideck/job/show/826e88dc-926b-415f-a724-a3a847aae3f2","scheduled":false,"scheduleEnabled":true,"enabled":true}]
```